### ISSUE TITLE
LLVM ERROR: SHT_NOBITS section '.bss' cannot have non-zero initializers

### ISSUE THREAD

**Issue by DutchGhost** [rust-lang/rust#81628]
I tried this code:

```rust
#[link_section = ".bss"]
pub static __: u8 = 1;

fn main() {
}
```

I expected to see this happen: This program should either compile, or only rustc should emit an error

Instead, this happened: On top of a rustc error, I also got an `LLVM ERROR`

### Meta

```
1.49
1.50.0-beta.8
1.51.0-nightly
```

<details><summary>Backtrace</summary>
<p>

```
 Compiling playground v0.0.1 (/playground)
LLVM ERROR: SHT_NOBITS section '.bss' cannot have non-zero initializers
error: could not compile `playground`

To learn more, run the command again with --verbose.
```

</p>
</details>
